test/test_darknet.cpp

```diff
@@ -0,0 +1,42 @@
+//****************************************************************************
+// SMaLL, Software for Machine Learning Libraries
+// Copyright 2023 by The SMaLL Contributors, All Rights Reserved.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// For additional details (including references to third party source code and
+// other files) see the LICENSE file or contact permission@sei.cmu.edu. See
+// Contributors.txt for a full list of contributors. Created, in part, with
+// funding and support from the U.S. Government (see Acknowledgments.txt file).
+// DM23-0126
+//****************************************************************************
+
+//#include <LAGraph.h>
+typedef float dtype;
+#include <acutest.h>
+
+#include <small.h>
+#include <small/models/Darknet.hpp>
+
+//****************************************************************************
+void test_darknet_parser(void)
+{
+    
+    std::string cfg = "../test/cfg_data/no_wspace_yolov3-tiny.cfg";
+    std::string weights = "../test/cfg_data/yolov3-tiny.weights";
+
+    // std::string cfg = "/home/nicholai/prog/cnn_cfgs/no_wspace_yolov3.cfg";
+    // std::string weights = "/home/nicholai/prog/yolo_weights/yolov3.weights";
+
+    std::cout << "\n\nReading Darknet CFG from " << cfg << std::endl;
+    std::cout << "Reading Darknet weights from " << weights << std::endl;
+
+    small::Darknet<small::FloatBuffer> model(cfg, weights);
```

Comment:
Put this within a try block. write one or more catch clauses for the various exceptions and reporting messages and TEST_ASSERT(false);

Reply:
I created the try block in [f419425](https://github.com/CMU-SPEED/SMaLLFramework/pull/7/commits/f4194253ec542140623dcae1d5440638c1b3ee99) but I only catch a general exception at the moment. I will leave this conversation open until I handle all the different possibilities.